src/MIDebugEngine/Engine.Impl/Variables.cs

```diff
@@ -1,15 +1,15 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using MICore;
```

Comment:
To address:

> 2. What is meant by "the value of the capture"? Is this result with the opening and closing parentheses removed (in the case of the second regex)?

Replace s_validAddressFormats with:
`static readonly Lazy<Regex> s_addressPattern = new Lazy<Regex>(new Regex(@"^(0x[0-9a-fA-F]+)\b"));`

Replace:
```
                if (Regex.IsMatch(result, s_validAddressFormats[i]))
                {
                    return result;
                }
```

With:
```
            Match m = s_addressPattern.Value.Match(result);
            if (m.Success)
            {
                return m.Captures[0]
            }
```

Reply:
To give a bit more of an explanation - in regex language '(<something-here>)' is a 'capture group'. Whatever you put there is available in the 'Captures' collection of a successful regex match.

---
In reply to: [592817678](https://github.com/microsoft/MIEngine/pull/1117#discussion_r592817678) [](ancestors = 592817678)